### PR TITLE
Feat: delivery button

### LINKED ISSUE #64: Add Delivery Location Button
## 📘 Description

A button should be added to the header allowing users to select the delivery country. Initially, the only available country will be **Costa Rica**, which will be displayed in a select within a modal. This button should be accompanied by an icon and allow users to view the selected country. When the button is clicked, a modal will open with a **select** component from **shadcn/ui** displaying the option for **Costa Rica** along with its flag. The modal can also contain any other information you deem relevant. The design will be **responsive** to ensure proper functionality across different devices.

## ✅ Acceptance Criteria

- The **"Deliver to Costa Rica"** button is located in the header and contains an icon.
- When clicked, the button opens a modal with a select showing only **Costa Rica** with its flag.
- The select is based on the **shadcn/ui** component.
- The design is **responsive**.

## 🌎 References

- [Lucide React Icons](https://lucide.dev/icons/)
- [shadcn/ui Select Component](https://ui.shadcn.com/docs/components/select)

## 📜 Additional Notes
- Blocked by: #54 

⚠ Do not apply until the ODHack begins.

⚠ Read our guidelines before applying.

### PR DESCRIPTION
**Pull request by zleypner**
# 📝 Add Delivery Location Button

## 🛠️ Issue
- Closes #64

## 📖 Description
This pull request implements the "Deliver to Costa Rica" button in the header. The button includes an icon and allows users to open a modal for selecting the delivery country. Currently, Costa Rica is the only option available, with its flag displayed in the selection modal. The design ensures responsiveness across different devices.

## ✅ Changes Made
- Added a "Deliver to Costa Rica" button in the header.
- Integrated a modal containing a select component based on `shadcn/ui`.
- Included the flag of Costa Rica in the select options.
- Ensured responsive design for both the button and modal.
- Incorporated the `Lucide React Icons` for the button.

## 🖼️ Media (Screenshots/Videos)
image

## 📜 Additional Notes
- The modal is designed to support future expansion for multiple delivery countries.
- Ensure consistent styling with the rest of the application.
- References:
- [Lucide React Icons](https://lucide.dev/)
- [shadcn/ui Select Component](https://ui.shadcn.dev/)
- [OpenMoji Emoji Library](https://openmoji.org/library/emoji-1F1E8-1F1F7/)

Please review the implementation and suggest any changes or improvements.

### CODE REVIEW

**Comment by zleypner** [comment]
Hello @derianrddev!
🚧 **Work in Progress**: I am working on this issue and will update with implementation progress ASAP ⚡

**Comment by zleypner** [comment]
Hello @derianrddev  ! Almost done, just checking last details. As per our conversation I will follow the example provided, looking for flag react icons to make easier adding other countries.

image

Example provided: 
image

**Comment by derianrddev** [comment]
@zleypner Great! Thanks for the update. Let me know if you need anything else to keep moving forward.

**Review by danielcdz — APPROVED** [review]

**Comment by zleypner** [comment]
👋 Hello @derianrddev ! 🚩 The flag has been added. If there’s anything else I may help with, please let me know.  I’ll be glad to proceed with the updates! 

image

File: frontend/app/components/header/Header.tsx [hunk 1]
```diff
@@ -118,4 +132,4 @@ export default function Header() {
       </header>
     </>
   );
-}
+}
```
Comment (derianrddev) [inline frontend/app/components/header/Header.tsx]: Add a line at the end of the file, please

File: frontend/app/components/ui/delivery-button.tsx
Comment (derianrddev) [inline frontend/app/components/ui/delivery-button.tsx:1]: Remove this file as it is unnecessary.

File: frontend/app/components/ui/delivery-location-button.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+              alt="Costa Rica flag"
+              className="inline-block w-6 h-6 ml-2" 
+            />
+          </SelectItem>
+      </SelectContent>
+    </Select>
+  )
+}
+
+export default DeliveryLocationButton
```
Comment (derianrddev) [inline frontend/app/components/ui/delivery-location-button.tsx]: Add a line at the end of the file, please

**Review by derianrddev — CHANGES_REQUESTED** [review]
@zleypner  Good job!  Just some minor changes.

**Comment by zleypner** [comment]
Hello @derianrddev ! I will proceed with the updates asap

**Review by derianrddev — APPROVED** [review]

**Review by danielcdz — APPROVED** [review]

**Pull request merged**